src/go/rpk/pkg/cli/cmd/cluster/config/lint.go

```diff
@@ -0,0 +1,115 @@
+// Copyright 2022 Vectorized, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+package config
+
+import (
+	"path/filepath"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+	"github.com/vectorizedio/redpanda/src/go/rpk/pkg/api/admin"
+	"github.com/vectorizedio/redpanda/src/go/rpk/pkg/config"
+	"github.com/vectorizedio/redpanda/src/go/rpk/pkg/out"
+	"gopkg.in/yaml.v3"
+)
+
+func newLintCommand(fs afero.Fs) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "lint",
+		Short: "Remove any deprecated content from redpanda.yaml.",
+		Long: `Remove any deprecated content from redpanda.yaml.
+
+Deprecated content includes properties which were set via redpanda.yaml
+in earlier versions of redpanda, but are now managed via Redpanda's
+central configuration store (and via 'rpk cluster config edit').
+
+This command is not for general changes to cluster configuration: use this only
+when redpanda will not start due to a configuration issue.
+
+This command erases a named property from the local node's internal 
+cache, so that on next startup redpanda will treat the setting as if it 
+was set to the default.
+`,
+		Run: func(cmd *cobra.Command, propertyNames []string) {
+			p := config.ParamsFromCommand(cmd)
+			cfg, err := p.Load(fs)
+			out.MaybeDie(err, "unable to load config: %v", err)
+
+			client, err := admin.NewClient(fs, cfg)
+			out.MaybeDie(err, "unable to initialize admin client: %v", err)
+
+			schema, err := client.ClusterConfigSchema()
+			out.MaybeDie(err, "unable to query config schema: %v", err)
+
+			configFile, err := p.LocateConfig(fs)
+			// The LocateConfig error type is a full explanation, pass it through
+			// without qualification.
+			out.MaybeDie(err, "%v", err)
+
+			configIn, err := afero.ReadFile(fs, configFile)
+			out.MaybeDie(err, "unable to read config file %q: %v", configFile, err)
+
+			// Decode into intermediate yaml.Node representation to preserve comments
+			// when writing the linted file back out.
+			inputDoc := make(map[string]yaml.Node)
+			err = yaml.Unmarshal(configIn, inputDoc)
+			out.MaybeDie(err, "unable to parse config file: %v", err)
+
+			cleanedDoc := make(map[string]yaml.Node)
+			var cleanedProperties []string
+
+			for key, node := range inputDoc {
+				if key == "redpanda" {
+					var cleanedContent []*yaml.Node
+					for i := 0; i < len(node.Content); i += 2 {
+						keyNode := node.Content[i]
+						valNode := node.Content[i+1]
+						key := keyNode.Value
+						_, is_cluster_config := schema[key]
+						if is_cluster_config {
```

Comment:
```suggestion
						if _, exists := schema[key]; exists {
```
but if not this suggestion, `is_cluster_config` => `isClusterConfig`

Reply:
One day I'm going to open a Go PR without a snake case variable, one day.  Changed to bumpyCase.